### PR TITLE
Add loans

### LINKED ISSUE #11: Add loans
- add loans to docker-compose
- create loans migration
- create payments migration

### PR DESCRIPTION
**Pull request by BarracudaPff**
- [x] add loans to docker-compose
- [x] create loans migration
- [x] create payments migration
closes: #11

### CODE REVIEW

File: sql/20190428223823_create_loans_table.sql [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    "client_id"            INT     NOT NULL,
+    "application_id"       INT     NOT NULL,
+    "amount"               INT     NOT NULL,
+    "term"                 INT     NOT NULL,
+    "interest"             INT     NOT NULL,
+    "currency"             TEXT    NOT NULL,
+    "date"                 TEXT    NOT NULL,
+    "completed"            BOOLEAN NOT NULL,
+    "rounding"             INT     NOT NULL,
+    "regularPaymentAmount" INT     NOT NULL,
```
Comment (malinink) [inline sql/20190428223823_create_loans_table.sql]: use underscore here

File: sql/20190428223823_create_loans_table.sql [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+CREATE TABLE "loans" (
+    "id"                   SERIAL  NOT NULL,
+    "client_id"            INT     NOT NULL,
+    "application_id"       INT     NOT NULL,
+    "amount"               INT     NOT NULL,
+    "term"                 INT     NOT NULL,
+    "interest"             INT     NOT NULL,
+    "currency"             TEXT    NOT NULL,
+    "date"                 TEXT    NOT NULL,
+    "completed"            BOOLEAN NOT NULL,
```
Comment (malinink) [inline sql/20190428223823_create_loans_table.sql]: finished ?

File: sql/20190428223823_create_loans_table.sql [hunk 3]
```diff
@@ -0,0 +1,23 @@
+-- migrate:up
+CREATE TABLE "loans" (
+    "id"                   SERIAL  NOT NULL,
+    "client_id"            INT     NOT NULL,
+    "application_id"       INT     NOT NULL,
+    "amount"               INT     NOT NULL,
+    "term"                 INT     NOT NULL,
+    "interest"             INT     NOT NULL,
+    "currency"             TEXT    NOT NULL,
+    "date"                 TEXT    NOT NULL,
```
Comment (malinink) [inline sql/20190428223823_create_loans_table.sql]: use timestamp

File: sql/20190428223823_create_loans_table.sql [hunk 4]
```diff
@@ -0,0 +1,23 @@
+-- migrate:up
+CREATE TABLE "loans" (
+    "id"                   SERIAL  NOT NULL,
+    "client_id"            INT     NOT NULL,
+    "application_id"       INT     NOT NULL,
+    "amount"               INT     NOT NULL,
```
Comment (malinink) [inline sql/20190428223823_create_loans_table.sql]: bigint

File: sql/20190428223823_create_loans_table.sql [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+    "term"                 INT     NOT NULL,
+    "interest"             INT     NOT NULL,
+    "currency"             TEXT    NOT NULL,
+    "date"                 TEXT    NOT NULL,
+    "completed"            BOOLEAN NOT NULL,
+    "rounding"             INT     NOT NULL,
```
Comment (malinink) [inline sql/20190428223823_create_loans_table.sql]: add check 1,10, 100

File: sql/20190428223823_create_loans_table.sql [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+    "regularPaymentAmount" INT     NOT NULL,
+    "remainingTerm"        INT     NOT NULL,
+    PRIMARY KEY (id),
+    FOREIGN KEY (client_id) REFERENCES clients (id) ON DELETE CASCADE ON UPDATE CASCADE,
```
Comment (malinink) [inline sql/20190428223823_create_loans_table.sql]: use RESTRICT!

File: sql/20190428225137_create_payments_table.sql [hunk 7]
```diff
@@ -0,0 +1,16 @@
+-- migrate:up
+CREATE TABLE "payments" (
+    "id"                 SERIAL NOT NULL,
+    "date"               TEXT   NOT NULL,
+    "amount"             INT    NOT NULL,
+    "percent"            INT    NOT NULL,
+    "body"               INT    NOT NULL,
+    "type"               TEXT   NOT NULL,
+    "remainCreditBody"   INT    NOT NULL,
```
Comment (malinink) [inline sql/20190428225137_create_payments_table.sql]: underscore

File: sql/20190428225137_create_payments_table.sql [hunk 8]
```diff
@@ -0,0 +1,16 @@
+-- migrate:up
+CREATE TABLE "payments" (
+    "id"                 SERIAL NOT NULL,
+    "date"               TEXT   NOT NULL,
+    "amount"             INT    NOT NULL,
```
Comment (malinink) [inline sql/20190428225137_create_payments_table.sql]: actually all money data types should be bigint

File: sql/20190428225137_create_payments_table.sql [hunk 9]
```diff
@@ -0,0 +1,16 @@
...
+    "percent"            INT    NOT NULL,
+    "body"               INT    NOT NULL,
+    "type"               TEXT   NOT NULL,
+    "remainCreditBody"   INT    NOT NULL,
+    "fullEarlyRepayment" INT    NOT NULL,
+    CONSTRAINT type_is_correct CHECK ("type" IN ('regular', 'early')),
```
Comment (malinink) [inline sql/20190428225137_create_payments_table.sql]: should be named `payments_type_check`

**Review by malinink — CHANGES_REQUESTED** [review]

File: sql/20190428223823_create_loans_table.sql (hunk 2 above)
Comment (BarracudaPff) [reply sql/20190428223823_create_loans_table.sql]: We discussed it in the specification and "completed" was fine, also in loans, we use "completed"

File: sql/20190428223823_create_loans_table.sql (hunk 3 above)
Comment (BarracudaPff) [reply sql/20190428223823_create_loans_table.sql]: Should I fix from `TEXT` to `TIMESTAMP` in other tables?

Comment (malinink) [reply sql/20190428223823_create_loans_table.sql]: yeep!

File: sql/20190428223823_create_loans_table.sql (hunk 2 above)
Comment (malinink) [reply sql/20190428223823_create_loans_table.sql]: okey

File: sql/20190428225137_create_payments_table.sql [hunk 10]
```diff
@@ -0,0 +1,16 @@
+-- migrate:up
+CREATE TABLE "payments" (
+    "id"                  SERIAL    NOT NULL,
+    "date"                TIMESTAMP NOT NULL,
+    "amount"              BIGINT    NOT NULL,
+    "percent"             INT       NOT NULL,
+    "body"                BIGINT    NOT NULL,
+    "type"                TEXT      NOT NULL,
+    "remain_credit_body"  BIGINT    NOT NULL,
+    "fullEarly_repayment" BIGINT    NOT NULL,
```
Comment (malinink) [inline sql/20190428225137_create_payments_table.sql]: missed Camel here

**Review by malinink — CHANGES_REQUESTED** [review]

**Comment by BarracudaPff** [comment]
@levozavr 
Looks like there is a problem in clients. It's in birthDate. 
Perhaps due to the lack of parse in infrastructure model. `YYYY-MM-DD` converts to standard `YYYY-MM-DDThh:mm:ssZ`
For example in `GET http://localhost:8080/v1/12`:
```json
path: $, 
{
  "birthDate": "2019-01-02T00:00:00Z",
  "children": 0,
  "firstName": "TestUser",
  "jobs": [
    {
      "name": "test1",
      "wage": 72364
    },
    {
      "name": "test4",
      "wage": 67628
    }
  ],
  "lastName": "Testerov",
  "maritalStatus": "married",
  "passport": {
    "series": 1234,
    "number": 847881
  },
  "sex": "male"
}
expected: 
{
  "birthDate": "2019-01-02",
  "firstName": "TestUser",
  "lastName": "Testerov",
  "passport": {
    "series": 1234,
    "number": 847881
  },
  "jobs": [
    {
      "name": "test1",
      "wage": 72364
    },
    {
      "name": "test4",
      "wage": 67628
    }
  ],
  "sex": "male",
  "maritalStatus": "married"
}
, reason: all key-values did not match
```

File: sql/20190505185912_change_text_to_timestamp_clients_table.sql [hunk 11]
```diff
@@ -0,0 +1,3 @@
+-- migrate:up
+ALTER TABLE clients ALTER COLUMN "birth_date" TYPE DATE
+    USING to_date("birth_date", 'YYYY-MM-DD');
```
Comment (levkovalenko) [inline sql/20190505185912_change_text_to_timestamp_clients_table.sql]: @BarracudaPff 'cause here added new migration, clients don't wok with DATE at this version. Clients use TEXT.

Comment (BarracudaPff) [reply sql/20190505185912_change_text_to_timestamp_clients_table.sql]: ok

**Review by malinink — APPROVED** [review]

**Pull request merged**